src/query/formats/src/field_decoder/nested.rs

```diff
@@ -318,6 +322,19 @@ impl NestedValues {
         Ok(())
     }
 
+    fn read_geometry<R: AsRef<[u8]>>(
+        &self,
+        column: &mut BinaryColumnBuilder,
+        reader: &mut Cursor<R>,
+    ) -> Result<()> {
+        let mut buf = Vec::new();
+        self.read_string_inner(reader, &mut buf)?;
+        let geom = Ewkb(buf).to_ewkb(CoordDimensions::xy(), None)?;
```

Comment:
The values inserted by users should be in text format EWKT, we should parse the EWKT string and then convert it to binary format EWKB.